soroban-env-host/observations.json

```diff
@@ -0,0 +1,1091 @@
+    "  29 leave VM:6d982c6:sym#189 -> Err(Error(Contract, #12345))": "cpu:11284703, mem:5448961, prngs:-/d8a09d57, objs:5@83222790/158@2fe130a3, vmem:1114112@1a8c6508, evt:3@a4578627, store:1@61d62dd9/11@c23283c6",
+    "  30 ret try_call -> Ok(\"Error(Contract, #12345)\")": "cpu:11284703, mem:5448961, prngs:-/d8a09d57, objs:4@8293cae0/158@2fe130a3, vmem:-, evt:3@b3315b9, store:-/8@7533dcbf",
+    "  31 leave VM:4f7f2d17:invoke -> Ok(Void)": "cpu:11284739, mem:5448961, prngs:-/d8a09d57, objs:4@8293cae0/158@2fe130a3, vmem:1114112@95b858fb, evt:3@b3315b9, store:-/8@7533dcbf",
+    "  32 enter SAC:baaaf5fd:balance(Address(obj#27))": "cpu:11367622, mem:5454115, prngs:-/d8a09d57, objs:-/160@ec6e58c9, vmem:-, evt:3@b3315b9, store:-/8@7533dcbf",
+    "  33 leave SAC:baaaf5fd:balance -> Ok(I128(100000000))": "cpu:11457455, mem:5457574, prngs:-/d8a09d57, objs:-/168@3a413db5, vmem:-, evt:3@b3315b9, store:-/8@7533dcbf",
+    "  34 enter SAC:baaaf5fd:balance(Address(obj#77))": "cpu:11540338, mem:5462728, prngs:-/d8a09d57, objs:-/170@c02732b5, vmem:-, evt:3@b3315b9, store:-/8@7533dcbf",
+    "  35 leave SAC:baaaf5fd:balance -> Ok(I128(0))": "cpu:11602956, mem:5465765, prngs:-/d8a09d57, objs:-/173@7c08eb34, vmem:-, evt:3@b3315b9, store:-/9@4e4e207f",
+    "  36 enter VM:6d982c6:sym#347()": "cpu:14011395, mem:6816606, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:1114112@bfc9a054, evt:3@b3315b9, store:-/9@4e4e207f",
+    "  37 call has_contract_data(51035662, 2)": "cpu:14014631, mem:6816636, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:-, evt:3@b3315b9, store:-/9@4e4e207f",
+    "  38 ret has_contract_data -> Ok(\"False\")": "cpu:14017581, mem:6816636, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:-, evt:3@b3315b9, store:0@58c79e45/9@4e4e207f",
+    "  39 call has_contract_data(51035662, 1)": "cpu:14017581, mem:6816636, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:-, evt:3@b3315b9, store:0@58c79e45/9@4e4e207f",
+    "  40 ret has_contract_data -> Ok(\"False\")": "cpu:14030492, mem:6817468, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:-, evt:3@b3315b9, store:0@58c79e45/10@7d7a448f",
+    "  41 call has_contract_data(51035662, 0)": "cpu:14030492, mem:6817468, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:-, evt:3@b3315b9, store:0@58c79e45/10@7d7a448f",
+    "  42 ret has_contract_data -> Ok(\"False\")": "cpu:14044165, mem:6818368, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:-, evt:3@b3315b9, store:0@58c79e45/11@1bea8adc",
+    "  43 leave VM:6d982c6:sym#347 -> Ok(False)": "cpu:14044165, mem:6818368, prngs:-/d8a09d57, objs:0@150c532c/176@9ba855af, vmem:1114112@d3636605, evt:3@b3315b9, store:0@58c79e45/11@1bea8adc",
+    "  44 end": "cpu:14065967, mem:6819900, prngs:-/d8a09d57, objs:-/178@dcadce48, vmem:-, evt:3@b3315b9, store:-/11@1bea8adc"
+  },
+  "test::vec::empty_vec_back": {
```

Comment:
Why do we not have host function call traces here?

Reply:
They're direct (rust) calls on the host, not mediated through the VM->host dispatch machinery where I inserted hooks. I might be able to insert more hooks in the VmCallerEnv emulation layer, that's a good point. I'll try that too.

Reply:
Yeah, it would be really nice to intercept any host call, because we rarely dispatch test host calls through VM.

Reply:
done.